### ISSUE TITLE
code of meta-learning when updating model's param in inner and outer update

### ISSUE THREAD

**Issue by zzowenzz** [HaowenBai/TDFusion#2]
Hi @HaowenBai , congrats to the publication. Just would like to ask 2 questions related to the model updating which I believe is the core of this paper and meta learning.

1. Should you call `inner_update()` twice to separately update fusion net and task net, giving that both modules are initialized differently that own unique optimizer and lr scheduler? Currenetly there is only 1 function calling which is after fusion net update but before task net update in the inner update stage. 
2. In both `inner_update()` and `outer_update()`, you pass `lr_epoch` as the second argument, which reflects the lr of the passing module. Since `lr_epoch=optimizer1.param_groups[0]['lr']` only reflects the optimizer1's lr (which is for fusion net), shall you pass different optimizer in inner and outer update?

**Comment by HaowenBai** [comment]
> Hi [@HaowenBai](https://github.com/HaowenBai) , congrats to the publication. Just would like to ask 2 questions related to the model updating which I believe is the core of this paper and meta learning.
> 
> 1. Should you call `inner_update()` twice to separately update fusion net and task net, giving that both modules are initialized differently that own unique optimizer and lr scheduler? Currenetly there is only 1 function calling which is after fusion net update but before task net update in the inner update stage.
> 2. In both `inner_update()` and `outer_update()`, you pass `lr_epoch` as the second argument, which reflects the lr of the passing module. Since `lr_epoch=optimizer1.param_groups[0]['lr']` only reflects the optimizer1's lr (which is for fusion net), shall you pass different optimizer in inner and outer update?

1 `inner_update()` updates are applied only to F because the `outer_update` requires the computation graph of F’ with respect to F, necessitating the retention of both F and F’ in the model. However, the `outer_update` only requires T’, not T. Therefore, we simplified this step to involve saving and loading model T, as indicated in lines 128 and 196 of `train.py`. Before updating T to T’, we store parameters of T, and after using T’ for the outer update, we reload T. This approach also has the advantage of not requiring modifications to the downstream task network code, unlike fusion networks, making it more convenient and efficient.

2 In our code, the learning rates of the three optimizers are synchronized and always have the same value. Therefore, it doesn’t matter which one you input. However, using the learning rate of optimizer 3 might be more standard.

**Comment by zzowenzz** [comment]
Thanks for your update.
